tidb-cloud/tidb-cloud-dm-precheck-and-troubleshooting.md

```diff
@@ -0,0 +1,106 @@
+- Amazon RDS: the process is similar to setting the `binlog_format`. The only difference is that the parameter you change is `max_connections` instead of `binlog_format`.  See [Configuring MySQL binary logging](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.MySQL.BinaryFormat.html).
+- MySQL: configure `max_connections` following the document [max_connections](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_connections).
+
+If the error occurs in the TiDB Cloud cluster, configure `max_connections` following the document [max_connections](https://docs.pingcap.com/tidb/stable/system-variables#max_connections).
+
+## Migration errors and solutions
+
+This section describes the problems and solutions you might encounter during the migration. These error messages are shown on the **Migration Job Details** page.
+
+### Error message: "The required binary log for migration no longer exists on the source database. Please make sure binary log files are kept for long enough time for migration to succeed."
+
+This error means that the binlogs to be migrated has been cleaned up and can only be restored by creating a new task.
+
+Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs if it's needed by some migration job.
+
+### Error message: "Failed to connect to the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
+
+This error means that the connection to the source database failed. Check whether the source database is started and can be connected to using the specified parameters. After confirming that the source database is available, you can try to recover the task by clicking **Restart**.
+
+### The migration task is interrupted and contains the error "driver: bad connection" or "invalid connection"
+
+This error means that the connection to the downstream TiDB cluster failed. Check whether the downstream TiDB cluster is in the `normal` state and can be connected with the username and password specified by the job. After confirming that the downstream TiDB cluster is available, you can try to resume the task by clicking **Restart**.
+
+### Error message: "Failed to connect to the TiDB cluster using the given user and password. Please make sure TiDB Cluster is up and can be connected to using the given user and password."
+
+Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by clicking **Restart**.
```

Comment:
The "normal" state is now updated to the "Available" state.

Reply:
`normal` includes `Available`, Modifying(scale in/out)

maybe add this detail in `()`

Reply:
Updated: It is recommended to check whether the TiDB cluster is in a normal state (including `Available` and `Modifying`).